core/src/main/java/amino/run/oms/migrationdecision/MetricWatcher.java

```diff
@@ -222,55 +276,75 @@ private long aggregatedProcessingTime(ReplicaID replicaID)
                 microServiceManager.getMicroServiceMetric(replicaID);
 
         // calculate time factor impact on RPC with latency and bandwidth
-        long averageExecutionTime = 0;
-        long summation = 0;
-        long data;
+        long totElapsedTime = 0;
+        long rttSum = 0;
+        long dataSize;
         NodeMetric nodeMetric;
+        Map<InetSocketAddress, NodeMetric> ksMetric;
         for (Map.Entry<InetSocketAddress, RPCMetric> replicaMetric : replicaMetrics.entrySet()) {
-            averageExecutionTime += replicaMetric.getValue().elapsedTime;
+            totElapsedTime += replicaMetric.getValue().elapsedTime;
 
-            data = replicaMetric.getValue().dataSize;
-            nodeMetric = replicaNodeMetric.get(replicaMetric.getKey());
-            // kernel client utilized by micro service but not running in kernel server are
-            // ignored in process time computation because no metric data available of kernel server
-            // from there.
-            if (nodeMetric == null) {
-                logger.info("skipping kernel client not part of kernel server");
+            dataSize = replicaMetric.getValue().dataSize;
+            // kernel server metrics for which RPC data is getting processed
+            try {
+                ksMetric = serverManager.getKernelServerMetric(replicaMetric.getKey());
+            } catch (KernelServerNotFoundException e) {
+                // typical scenario like end user application(kvstore, minnieTwitter) where kernel
+                // client is utilized by micro service
+                // but kernel server do not run. In such scenario RPC metrics are collected against
+                // kernel client
+                // but as the client is not part of kernel server so no node metric are collected.
+                // Hence ignoring time computation
+                logger.info(
+                        String.format(
+                                "skipping kernel client [%s] not part of kernel server",
+                                replicaMetric.getKey()));
                 continue;
             }
 
-            summation += (data / nodeMetric.rate + nodeMetric.latency);
+            nodeMetric = ksMetric.get(currentKernelServer);
+            rttSum += (dataSize / nodeMetric.rate + nodeMetric.latency);
         }
 
+        long avgElapsedTime = 0;
+        long avgRtt = 0;
         if (!replicaMetrics.isEmpty()) {
-            averageExecutionTime = averageExecutionTime / replicaMetrics.size();
+            avgElapsedTime = totElapsedTime / replicaMetrics.size();
+            avgRtt = rttSum / replicaMetrics.size();
         }
 
-        return averageExecutionTime + summation;
+        return avgElapsedTime + avgRtt;
```

Comment:
This seems wrong.  Don't you want to *subtract* the RTT from the elapsedTime to get the processing (computation) time?  Also, a single inbound RPC may result in multiple "dowstream" outbound RPC's, some in serial, and some in parallel.  Using the avgRtt here (even assuming your substract it as mentioned above) implies a single outbound RPC per inbound RPC, which is not correct?  I think you need to keep track of the ratio of inbound:outbound (say 1:n) and subtract n times the avgRtt?